crates/via-serve-static/src/lib.rs

```diff
@@ -0,0 +1,141 @@
+mod etag;
+mod respond;
+mod static_file;
+mod stream_file;
+
+use bitflags::bitflags;
+use std::{path::Path, sync::Arc};
+use via::{Endpoint, Error, Result};
+
+use crate::respond::{respond_to_get_request, respond_to_head_request};
+
+pub struct ServeStatic<'a, State> {
+    eager_read_threshold: u64,
+    read_stream_timeout: u64,
+    endpoint: Endpoint<'a, State>,
+    flags: Flags,
+}
+
+#[derive(Clone)]
+pub(crate) struct ServerConfig {
+    eager_read_threshold: u64,
+    read_stream_timeout: u64,
+    path_param: &'static str,
+    public_dir: Arc<Path>,
+    flags: Flags,
+}
+
+bitflags! {
+    #[derive(Debug, Clone, Copy)]
+    pub(crate) struct Flags: u8 {
```

Comment:
I like the way we handle config options with `bitflags`. Actix does the same and I admire their static server. We'll definitely keep this in the rewrite.